### PR TITLE
[Term Entry] D3 Interactivity .transition() added

### LINKED ISSUE #4032: [Term Entry] D3 Interactivity .transition()
### Reason/inspiration (optional)

We would like a new term entry on the `.transition()` method for [D3](https://www.codecademy.com/resources/docs/d3): Interactivity. The entry should go in a new file under `docs/content/d3/concepts/interactivity/terms/transition/transition.md`.

The entry should describe what the method does. Be sure to include the following sections:

- A "Syntax" section with a ` ```pseudo` code snippet.
- An "Example" section with a ` ```js `code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by antriksh-9**
### Description

I was assigned this [issue](https://github.com/Codecademy/docs/issues/4032).
Created Codecademy Docs term entry on the .transition() method for D3.
Closes #4032.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/d3/concepts/interactivity/terms/transition/transition.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  - 'Animations'
+  - 'Selection'
+  - 'Transitions'
+CatalogContent:
+  - 'learn-d3'
+  - 'paths/web-development'
+---
+
+In D3.js, the **`.transition()`** method is used to create a smooth transition between different states of a selection. Transitions provide a way to animate changes in the DOM elements over time.
+It allows to smoothly update the appearance of elements making visualizations more interactive and engaging.
```
Comment (dakshdeepHERE) [inline content/d3/concepts/interactivity/terms/transition/transition.md:18]:
```suggestion
In D3.js, the **`.transition()`** method is used to create a smooth transition between different states of a selection. Transitions provide a way to animate changes in the DOM elements over time. It allows to smoothly update the appearance of elements making visualizations more interactive and engaging.
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
Only one change from my side

**Comment by antriksh-9** [comment]
Hey @dakshdeepHERE I have made the changes.

**Review by dakshdeepHERE — APPROVED** [review]
Looks good for second review.

**Review by avdhoottt — APPROVED** [review]
Looks good to get merged.

**Pull request merged**